src/engine/Join.cpp

```diff
@@ -172,6 +172,24 @@ Result Join::computeResult(bool requestLaziness) {
   return lazyJoin(std::move(leftRes), std::move(rightRes), requestLaziness);
 }
 
+// _____________________________________________________________________________
+std::optional<std::shared_ptr<QueryExecutionTree>>
+Join::setPrefilterGetUpdatedQueryExecutionTree(
+    std::vector<PrefilterVariablePair> prefilterVariablePairs) const {
+  auto optNewSubtreeLeft = _left->setPrefilterGetUpdatedQueryExecutionTree(
+      prefilterExpressions::detail::makePrefilterExpressionVecCopy(
+          prefilterVariablePairs));
+  auto optNewSubtreeRight = _right->setPrefilterGetUpdatedQueryExecutionTree(
+      std::move(prefilterVariablePairs));
+  if (!optNewSubtreeLeft.has_value() && !optNewSubtreeRight.has_value()) {
+    return std::nullopt;
+  }
+  return ad_utility::makeExecutionTree<Join>(
+      getExecutionContext(), optNewSubtreeLeft.value_or(_left->clone()),
+      optNewSubtreeRight.value_or(_right->clone()), _leftJoinCol,
+      _rightJoinCol);
```

Comment:
I am not sure whether it is necessary to `clone()` here, especially during query planning.
I think the correct pattern is to only `clone` when we actually modify a tree (which we don't).